### DISCUSSION TITLE
Discussion on June 20

### DISCUSSION THREAD

**Discussion by libertyeagle** [phoenix-dataplane/phoenix#136] [General]
TODO items:
1. dynamic binding. two parts: codegen for backend's dynamic dispatch code; dynamically loading of generated code.  
2. we need a scheduler for small messages to combine multiple SgEs.
3. centralized policy: problem definition & evaulation scenarios. We should demonstrate 2-3 cases that centralized policy can greatly improve performance, using some simple heuristics

**Comment by crazyboycjr** [comment]
Best to avoid using the name WrIdentifier (a RDMA term in RPC logic)
ptr_app <-> ptr_backend
ShmView <-> RpcMessage

**Reply by crazyboycjr** [reply]
ReclaimRecvBuf <-> SendCompletion

**Reply by crazyboycjr** [reply]
SharedRecvBuffer <-> SharedHeapRegion
